### PR TITLE
Add BLIS example

### PR DESCRIPTION
**Pull request by ken-matsui**

### CODE REVIEW

File: examples/BLIS.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    //
+    // __attribute__((noinline))
+    // void kernel(
+    //   uint8_t *restrict aa,
+    //   uint8_t *restrict ab,
+    //   uint8_t *restrict ac
+    // ) {
+    //   for (int ad = 0; ad < 16; ad++) {
+    //   for (int ae = 0; ae < 16; ae++) {
+    //     TODO: supposed to have let Bc = B.pack_into() here?
```
Comment (ken-matsui) [inline examples/BLIS.rs:145]: In my BLIS implementation and scheduled impl, we have packing here. But the generated code doesn't. I would like to know if this comes from just my misunderstanding, or if the implementation needs modification.

File: src/codegen/cpu.rs [hunk 2]
```diff
@@ -83,6 +83,7 @@ impl<'a, Tgt: Target<Level = CpuMemoryLevel>> CpuCodeGenerator<'a, Tgt> {
         writeln!(main_body_str, "}}")?;
 
         self.headers.emit(Tgt::target_id(), out)?;
+        out.write_str("\n")?;
```
Comment (ken-matsui) [inline src/codegen/cpu.rs:86]: This is for the newline between headers and the kernel.

File: src/pprint.rs [hunk 3]
```diff
@@ -108,7 +108,11 @@ fn pprint_inner<'a, Tgt, Aux>(
             }
             PrintMode::Compact => {
                 let joined = imp.aux().extra_column_values().join(", ");
-                r = row![format!("{indent_str}/* {joined} */\n{main_str}\n")];
+                if !joined.is_empty() {
+                    r = row![format!("{indent_str}/* {joined} */\n{main_str}\n")];
+                } else {
+                    r = row![main_str];
+                }
```
Comment (ken-matsui) [inline src/pprint.rs:115]: This is for omitting unnecessary `/* */` when there are no extra columns.

File: examples/BLIS.rs (hunk 1 above)
Comment (ken-matsui) [reply examples/BLIS.rs:145]: For packing, I tried `.move_param(1, CpuMemoryLevel::RF, row_major(2), None)` but didn't work.

```console
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /tmp/samkaufman/morello/src/scheduling_sugar.rs:72:10
```

Comment (ken-matsui) [reply examples/BLIS.rs:145]: For more details, please read my weekly notes.

File: src/codegen/cpu.rs (hunk 2 above)
Comment (samkaufman) [reply src/codegen/cpu.rs:86]: Thanks! If you break this into another PR, happy to merge into main immediately.

File: src/pprint.rs (hunk 3 above)
Comment (samkaufman) [reply src/pprint.rs:115]: As above, if you break this into another PR, happy to merge into main immediately.

**Pull request closed without merging**